metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -321,11 +322,20 @@ def extract_dbt_entities(
 
 
 def load_file_as_json(uri: str) -> Any:
-    if re.match("^https?://", uri):
-        return json.loads(requests.get(uri).text)
-    else:
-        with open(uri, "r") as f:
-            return json.load(f)
+   if re.match("^https?://", uri):
+       return json.loads(requests.get(uri).text)
+   elif re.match("^s3?://", uri):
+       path = uri.replace("s3://","").split("/")
+       bucket=path.pop(0)
```

Comment:
imo, mutations are harder to reason about in general.

We can get the bucket and key by just splitting the string once.

```python
s3_bucket, s3_key = uri.replace("s3://", "").split("/", 1)
# ^ split it to a maximum of once 


>>> uri = 's3://bucket/to/some/key'
>>> s3_bucket, s3_key = uri.replace("s3://", "").split("/", 1)
>>> s3_bucket
'bucket'
>>> s3_key
'to/some/key'
```